javascript/0441-arranging-coins.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * https://leetcode.com/problems/arranging-coins/
+ * Linear time
+ * Time O(n) | Space O(1)
+ * @param {number} n
+ * @return {number}
+ */
+var arrangeCoins = function(n) {
+    
+    let steps = 1;
+    let canBuild = 0;
+
+    while(n >= steps) {
+        n = n - steps;
+        canBuild++;
+        steps++;
+    }
+
+    return canBuild || 1;
+};
+
+/**
+ * Binary Search
+ * 
+ * Time O(n*log(n)) | Space O 
+ * @param {number} n
+ * @return {number}
+ */
+var arrangeCoins = function(n) {
+    
+    let left = 1;
+    let right = n;
+    let result = 0;
+  
+    while(left <= right) {
+  
+        const mid = Math.floor((right+left)/2);
+        const total = (1 + mid) * (mid/2);
+        if(n < total) {
+            right = mid -1;
+        } else {
+            left = mid+1;
+            result = Math.max(result, mid);
+        }
+    }
+  
+    return result;
+  };
+  
+/** 
+ * Math 
+ * Time O(1) | Space O(1)
+ * @param {number} n
+ * @return {number}
+ */
+var arrangeCoins = function(n) {
+    
+    let result1 = Math.floor((-1 + Math.sqrt(1+(8*n)))/2);
```

Comment:
Suggestion: Make the math operations more readable

```js
const value = Math.sqrt(1 + (8 * n)))
const add = ((value + -1) >> 1)
const sub = ((value - -1) >> 1)
```

Reply:
Done. Please check.